### DISCUSSION TITLE
Split ObjectModel into its own package without XF dependency?

### DISCUSSION THREAD

**Discussion by jcmanke** [xamarin/XamarinCommunityToolkit#1050] [Q&A] [answered]
I would love to use ObservableObject and AsyncCommand in my view models, but I also do not want my VMs to have a Xamarin.Forms dependency. Has there been any consideration of spinning off another NuGet package for the code in XCT which does not need to carry a dependency on Forms?

**Comment by pictos** [comment]
@jcmanke AFAIK there are no plans for releasing another package for ObservableObject. We remove the XF dependency from AsyncCommand, as you can see here #825 but if you see more dependencies, please open an issue as an enhacenment.

**Reply by AndreiMisiukevich** [reply]
I guess @jcmanke meant the dependency to Xamarin.Forms of Xamarin.CommunityToolkit (for example to be possible to use ObservableObject & AsyncCommand in a non-xamarin-forms project)

**Reply by jcmanke** [reply]
Exactly. Even in a Xamarin.Forms project, I prefer to keep my view models and my views in separate assemblies, and make the view models Xamarin.Forms agnostic. Otherwise it gets very easy to start violating MVVM principles.

**Reply by pictos** [reply]
OOh I see now, sorry for that. @jcmanke so not sure if we will do that, but thanks for the heads up we will see what we can do about it.

**Reply by mdstevensjr** [reply]
Looking for a separate package for the same reason as @jcmanke.

**Reply by JeanMarcMbouma** [reply]
> Looking for a separate package for the same reason as @jcmanke. 

This originated from @jamesmontemagno 's MVVM helpers library which is available as a nugget. Please give it a try and see if it works for your use case.

**Comment by pictos** [comment, marked as answer]
Hey friends, we will not change it here. On [MCT](https://github.com/CommunityToolkit/Maui) we don't have those objects anymore, you can find them on [CommunityToolkit.MVVM](https://github.com/CommunityToolkit/dotnet).

Thanks for the input and discussion. If you've any suggestion about it please open a discussion [here](https://github.com/CommunityToolkit/Maui/discussions/new)
